core/util.go

```diff
@@ -536,3 +538,28 @@ func LoadCert(filename string) (cert *x509.Certificate, err error) {
 	cert, err = x509.ParseCertificate(block.Bytes)
 	return
 }
+
+const retryJitter = 0.2
+
+// RetryBackoff calculates a backoff time based on number of retries, adapted from
+// https://github.com/grpc/grpc-go/blob/master/rpc_util.go#L311
+func RetryBackoff(retries int, base, max time.Duration, factor float64) time.Duration {
+	if retries == 0 {
+		return 0
+	}
+	backoff, fMax := float64(base), float64(max)
+	for backoff < fMax && retries > 1 {
+		backoff *= factor
+		retries--
+	}
```

Comment:
I think @bifurcation is proposing to replace the whole loop with a single call to `math.Pow`.

However, I suspect there's a good reason why grpc chose an explicit loop over `math.Pow`. One possibility: int math is faster than float math, and `math.Pow` expects floats. This may come into play for very high performance RPCs. Most likely our currently implementation isn't fast enough for the int/float difference to matter, but more saliently I think it's useful to keep a minimum difference relative to the implementation we are copying.